subprojects/docs/src/docs/userguide/upgrading_version_4.adoc

```diff
@@ -328,6 +377,149 @@ To override this behavior, you can explicitly declare the configuration to use i
+
+ * The http://checkstyle.sourceforge.net[Checkstyle] plugin now uses http://checkstyle.sourceforge.net/releasenotes.html#Release_8.12[8.12] instead of 6.19 by default.
+ * The http://codenarc.sourceforge.net[CodeNarc] plugin now uses https://github.com/CodeNarc/CodeNarc/blob/master/CHANGELOG.md#version-121-aug-2018[1.2.1] instead of 1.1 by default.
+ * The https://www.jacoco.org/jacoco/[JaCoCo] plugin now uses https://www.jacoco.org/jacoco/trunk/doc/changes.html[0.8.2] instead of 0.8.1 by default.
+ * The https://pmd.github.io/[PMD] plugin now uses https://pmd.github.io/pmd-6.8.0/pmd_release_notes.html#30-september-2018---680[6.8.0] instead of 5.6.1 by default.
++
+In addition, the default ruleset was changed from the now deprecated `java-basic` to `category/java/errorprone.xml`.
++
+We recommend configuring a ruleset explicitly, though.
+
+[[rel5.0:library_upgrades]]
+=== [5.0] Library upgrades
+Several libraries that are used by Gradle have been upgraded:
+
+ * Groovy was upgraded from 2.4.15 to http://groovy-lang.org/releasenotes/groovy-2.5.html[2.5.2].
+ * Ant has been upgraded from 1.9.11 to https://archive.apache.org/dist/ant/RELEASE-NOTES-1.9.13.html[1.9.13].
+ * The AWS SDK used to access S3 backed Maven/Ivy repositories has been upgraded from 1.11.267 to https://github.com/aws/aws-sdk-java/blob/master/CHANGELOG.md#111407-2018-09-11[1.11.407].
+ * The BND library used by the OSGi plugin has been upgraded from 3.4.0 to https://github.com/bndtools/bnd/wiki/Changes-in-4.0.0[4.0.0].
+ * The Google Cloud Storage JSON API Client Library used to access Google Cloud Storage backed Maven/Ivy repositories has been upgraded from v1-rev116-1.23.0 to v1-rev136-1.25.0.
+ * Ivy has been upgraded from 2.2.0 to http://ant.apache.org/ivy/history/2.3.0/release-notes.html[2.3.0].
+ * The JUnit Platform libraries used by the `Test` task have been upgraded from 1.0.3 to 1.3.1.
+ * The Maven Wagon libraries used to access Maven repositories have been upgraded from 2.4 to 3.0.0.
+ * SLF4J has been upgraded from 1.7.16 to https://www.slf4j.org/news.html[1.7.25].
+
+[[rel5.0:dependency_resolution]]
+=== [5.0] Fixes to dependency resolution
+
+Dependency resolutions fixes have been included in this release.
+By definition this could impact the set of resolved dependencies of your build.
+However the fixed issues are mostly about corner cases and combination with recent features and thus should have a limited impact.
+
+When a dependency constraint matched a real dependency, it was made part of the graph.
+However if for some reason the dependency was later evicted from the graph, the constraint remained present. Now, when the last non-constraint edge to a dependency disappears, all constraints for that dependency will be properly removed from the graph.
+
+[[rel5.0:changes_to_default_task]]
+=== [5.0] Changes to property factory methods on DefaultTask
+
+Property factory methods on DefaultTask are final
```

Comment:
I looks like it may have been a heading in its previous incarnation. As an introductory sentence, it shouldn't definitely say "...are _now_ final."

Also, monospace for type please.